### PR TITLE
added support for GWF 2014 and 2024 for brutal critical

### PR DESCRIPTION
**Pull request by dmportella**
known issue with versatile weapon is applying versatile brutal dmg to 1 handed dmg too will address this in a later commit

Pole arm attack examples with gwf 2024
Screenshot 2024-11-22 172202
Screenshot 2024-11-22 172227

polearm master 2014 with gwf 2014
image
image

pole strike 2024 + gwf 2014
image
image

long sword (versatile)
image
image

One handed should be using 1d8 for the brutal strike dmg and it is not (known issue)
image

GWF 2024 great axe
image

GWF 2024 long sword versatile
image

pole arm 2014 GWF 2024
image

pole strike 2024 GWF 2024
image

### CODE REVIEW

File: src/dndbeyond/base/utils.js [hunk 1]
```diff
@@ -283,19 +283,29 @@ async function buildAttackRoll(character, attack_source, name, description, prop
...
                     } else {
-                        // Apply great weapon fighting to brutal damage dice
-                        // TODO need to support 2024 version for GWF and Polearm Master (action name is pole strike now) and feats can be mixed
-                        if (character.hasGreatWeaponFighting(2014) &&
-                            ((properties["Attack Type"] == "Melee" &&
-                            ((properties["Properties"].includes("Versatile") && character.getSetting("versatile-choice") != "one") || properties["Properties"].includes("Two-Handed"))) ||
-                            name == "Polearm Master - Bonus Attack")) {
-                            brutal_dmg += "ro<=2"
+                        // check for both versions
+                        if (name.includes("Polearm Master") && (character.hasFeat("Polearm Master", false)) || properties["Attack Type"] == "Melee" && (properties["Properties"].includes("Versatile") || properties["Properties"].includes("Two-Handed"))) {
```
Comment (kakaroto) [inline src/dndbeyond/base/utils.js:287]: The statement is confusing, you basically have it like this:
```js
if (n && (p) || m && (v || t))
```
The `p`, or originally `(character.hasFeat("Polearm Master", false))` is between parenthesis on its own which confused me in trying to figure out the logic... (I thought "(hasPolearm || MelleeAttack)" which made no sense).
Then you also have a mix of `&&` and `||` that aren't grouped in parenthesis, so I can't quite figure out the order of operations here, or how to read it. I think it's likely to cause the wrong logic to execute.

File: src/dndbeyond/base/utils.js [hunk 2]
```diff
@@ -283,19 +283,29 @@ async function buildAttackRoll(character, attack_source, name, description, prop
...
+                                });
+                            }
+                        } else if (name.includes("Pole Strike") && (character.hasFeat("Polearm Master 2024", false)) || properties["Attack Type"] == "Melee" && (properties["Properties"].includes("Versatile") || properties["Properties"].includes("Two-Handed"))) {
+                            if (character.hasGreatWeaponFighting(2024)) {
+                                brutal_dmg = brutal_dmg.replace(/([0-9]*)d([0-9]+)([^\s+-]*)(.*)/g, (match, amount, faces, roll_mods, mods) => {
+                                    return new Array(parseInt(amount) || 1).fill(`1d${faces}${roll_mods}min3`).join(" + ") + mods;
+                                });
+                            } else if (character.hasGreatWeaponFighting(2014)) {
+                                    brutal_dmg = brutal_dmg.replace(/[0-9]*d[0-9]+/g, "$&ro<=2");
+                            }
```
Comment (kakaroto) [inline src/dndbeyond/base/utils.js:302]: Ideally this would be using the function I recommended creating in the other PR

**Review by kakaroto — CHANGES_REQUESTED** [review]
Some more changes requested here, thanks!

**Comment by dmportella** [comment]
Yep will resolve these np

**Comment by dmportella** [comment]
i will merge both brutal strike and brutal critical together and make one PR
so that we can use the same method with out having merge conflicts

**Comment by dmportella** [comment]
this has been merged into https://github.com/kakaroto/Beyond20/pull/1201

**Pull request closed without merging**